sos/plugins/juju.py

```diff
@@ -9,113 +9,201 @@
+                        # juju commands, clean them up and give them an
+                        # appropriate extension if possible
+                        filename = self._mangle_command(cmd)
+                        if re.match(r".*_--format_yaml.*", filename):
+                            filename = re.sub(r"_--format_yaml", "", filename)
+                            filename += ".yaml"
+                        self.add_cmd_output(
+                            "su - " + username + ' -c "' + cmd + '"',
+                            suggest_filename=filename,
+                        )
+
+                    # Get the model status as json so we can dig deeper
+                    status = self.get_cmd_output_now(
+                        "sudo -i -u {} juju status -m {} --format json".format(
+                            username, modelname
+                        ),
+                        suggest_filename="juju_status_-m_{}.json".format(
+                            modelname
+                        ),
+                    )
+                    self.collect_app_config(username, modelname, status)
+
+    # Get some high level juju configuration and start processing model info
+    def collect_juju_output(self, username):
+        # Run these commands once, no model or app option required
+        juju_cmds = [
+            "juju clouds --format yaml",
+            "juju controller-config --format yaml",
+            "juju storage --format yaml",
+            "juju storage-pools --format yaml",
+            "juju spaces --format yaml",
+            "juju credentials --format yaml",
+            "juju model-defaults --format yaml",
+            "juju show-controller --format yaml",
+        ]
+
+        # Get the model information as json so we can dig deeper
+        models = self.get_cmd_output_now(
+            "sudo -i -u {} juju models --format json".format(username),
+            suggest_filename="juju_models.json",
         )
-
-        for collection in collections:
+        self.collect_model_output(username, models)
+
+        # Use sudo to run the commands as the appropriate user and name the
+        # files something sane
+        for cmd in juju_cmds:
+            # The file names can be long and confusing from the juju commands,
+            # clean them up and give them an approriate extension if possible
+            filename = self._mangle_command(cmd)
+            if re.match(r".*_--format_yaml.*", filename):
+                filename = re.sub(r"_--format_yaml", "", filename)
+                filename += ".yaml"
             self.add_cmd_output(
-                "/usr/lib/juju/bin/mongoexport --ssl \
-                --dbpath=/var/lib/juju/db --db juju --collection {0} \
-                --jsonArray".format(collection),
-                suggest_filename="{}.json".format(collection))
+                "sudo -i -u {} {}".format(username, cmd),
+                suggest_filename=filename,
+            )
```

Comment:
Just nack. If the standard mangling isn't working for you, then let's have a conversation about that. Driving a truck through it left right and centre in these plugins is not a maintainable or reasonable way to approach that.